File: api/src/test/java/net/blueshell/api/factory/dto/request/MemberActivationRequestFactory.java
package net.blueshell.api.factory.dto.request;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.recovery.MemberActivationRequest;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for MemberActivationRequest test instances.
 */
@Component
@RequiredArgsConstructor
public class MemberActivationRequestFactory extends BaseDtoFactory<MemberActivationRequest> {

    @Override
    public Class<MemberActivationRequest> targetType() {
        return MemberActivationRequest.class;
    }

    @Override
    public MemberActivationRequest createBasic() {
        MemberActivationRequest dto = new MemberActivationRequest();
        dto.setToken(unique("tok"));
        dto.setUsername(unique("user"));
        dto.setPassword("Password123!");
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/request/MemberActivationRequestFactory.kt
package net.blueshell.api.factory.dto.request

import net.blueshell.api.dto.recovery.MemberActivationRequest
import net.blueshell.api.factory.dto.BaseDtoFactory
import org.springframework.stereotype.Component

/**
 * Factory for MemberActivationRequest test instances.
 */
@Component
class MemberActivationRequestFactory : BaseDtoFactory<MemberActivationRequest>() {

    override fun targetType(): Class<MemberActivationRequest> = MemberActivationRequest::class.java

    override fun createBasic(): MemberActivationRequest {
        val dto = MemberActivationRequest()
        dto.token = unique("tok")
        dto.username = unique("user")
        dto.password = "Password123!"
        return dto
    }
}
